polaris-core/src/main/java/org/apache/polaris/core/persistence/BasePersistence.java

```diff
@@ -408,4 +414,7 @@ boolean hasChildren(
   default BasePersistence detach() {
     return this;
   }
+
+  @Nonnull
+  PageToken.PageTokenBuilder<?> pageTokenBuilder();
```

Comment:
minor: Builders are usually quite simple collectors of programmatically collected arguments. Here it looks like this is more of a "parser". I'd personally prefer builders to be generated (immutables), and have custom parser factory interface.

Reply:
The intent here is that different metastores could have different PageToken implementations that they need to use. For example, it seems like EclipseLink may just not support pagination, which is fine. 

Practically, we need a way for persistence to be able to say "Ok, I fetched the first N results, here's a page token that will give you the next page". The builder which constructs that page token varies by persistence implementation. But if a different code structure would work better here I'm very open to it. Maybe it's more like a factory?